agent/consul/rate/handler.go

```diff
@@ -230,7 +230,13 @@ func (h *Handler) Allow(op Operation) error {
 		})
 
 		if enforced {
-			// TODO(NET-1382) - use the logger to print rate limiter logs.
+			h.logger.Warn("RPC blocked due to exceeded allowed rate limit",
```

Comment:
I did this work logging previously but I must have lost it in a merge conflict somewhere.

Reply:
Do we need this log line? We already have this one, which includes whether or not the limit was enforced:
https://github.com/hashicorp/consul/blob/b8a850c2a658c32666a5f42d3fef045feab57df8/agent/consul/rate/handler.go#L210-L215

Logging twice feels like it'll bloat the log (and drop buffer) unnecessarily. Although, we may want to increase the log-level of the other one.

Reply:
I added this because the comment was there to do so, but when I was thinking about this the questions/thoughts that I was thinking of (which I would love input on the answers are)?
- is trace the right level?
- if i am exceeding levels but its permissive, trace seems ok.  
- if I am dropping requests, I probably want info at a minimum and possibly warnings (IE, should I have to turn on tracing to know that my servers are blocking requests?)
- issuing warnings on the other hand for permissive might seem excessive?  may be not?
- should this be an
  - if enforced, issue a warning
  - else if permissive issue trace

Reply:
I think a warning log in both cases will be needed. The reasoning is that permissive mode will be used by operators as a mean to configure rate-limiting and having to activate the trace logs to do so would:
- Add to the operator burden to configure rate limiting
- Tracing would introduce an extra performance hit that could lead to requests being rate limited.

Reply:
Thanks @dhiaayachi and @boxofrad.  It seems like just warning once regardless of permissive vs enforcing is the way to go and the message contains limited=true/false?  This is what I did in https://github.com/hashicorp/consul/pull/15960/commits/0a98872947fa77bcb6ff443d97f92e25d720be20

Let me know what you think.